Objects/listobject.c

```diff
@@ -445,11 +445,19 @@ list_length(PyListObject *a)
 static int
 list_contains(PyListObject *a, PyObject *el)
 {
+    PyObject *item;
     Py_ssize_t i;
     int cmp;
 
-    for (i = 0, cmp = 0 ; cmp == 0 && i < Py_SIZE(a); ++i)
+    for (i = 0, cmp = 0 ; cmp == 0 && i < Py_SIZE(a); ++i) {
+        item = PyList_GET_ITEM(a, i);
+        if (item == el) {
```

Comment:
The compare code already has an early-out for an identity match.  No need to replicate that here.

Reply:
@rhettinger 

I expected this kind of performance improvement.
https://bugs.python.org/issue39425

but no effect for this case since this is not the same scenario that only needs one single same object.

Thanks for the review I've updated it :)